wallet/src/Pos/Wallet/Web/Methods/Redeem.hs

```diff
@@ -56,19 +55,17 @@ redeemAdaPaperVend
 redeemAdaPaperVend submitTx passphrase CPaperVendWalletRedeem {..} = do
     seedEncBs <- maybe invalidBase58 pure
         $ decodeBase58 bitcoinAlphabet $ encodeUtf8 pvSeed
-    aesKey <- either invalidMnemonic pure
-        $ deriveAesKeyBS <$> toSeed pvBackupPhrase
+    let aesKey = AesKey (mnemonicToAESKey pvBackupPhrase)
     seedDecBs <- either decryptionFailed pure
         $ aesDecrypt seedEncBs aesKey
     redeemAdaInternal submitTx passphrase pvWalletId seedDecBs
   where
     invalidBase58 =
         throwM . RequestError $ "Seed is invalid base58 string: " <> pvSeed
-    invalidMnemonic e =
-        throwM . RequestError $ "Invalid mnemonic: " <> toText e
```

Comment:
hm, this is potentially a regression.

Previous implementation would rise an error if client submitted backup phrase with illegal number of mnemonics. This rised an error and UI reacted on that error. I am not sure this is how it is done now though.

the old implementation rised an error if `fromMnemonic` returned left and your new implementation will create aesKey and then fail on seed decryption with a different error.  Not sure will this affect UI (or is it some sort of regression)

Running acceptance tests from Daedalus might be useful here.

Reply:
Note that the new implementation still does mnemonic validation. Actually, it's not possible to create a mnemonic that is invalid because the validation is done via the smart constructor. Therefore, if you hold a `Mnemonic` type _you know_ it's valid. That's also why the conversion functions are now pure and can't fail. 

You are raising a very valid point however. I didn't thought of Daedalus "pattern matching" on the error messages. That's probably something we want QA to be involved with. I'll ask.

Reply:
right, I have missed validation at json deserilization. My bad (althought we should still check for error msg format)

Reply:
After checking with @Nikola on the matter, Daedalus doesn't currently rely on any of the error messages that were triggered by the Mnemonic or BackupPhrase modules. 
I have opened a ticket [DDW-318](https://iohk.myjetbrains.com/youtrack/issue/DDW-318) to actually leverage the messages from the new implementation :+1: